File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrderSubtotalsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderSubtotalsRow } from "./OrderSubtotalsRow";

export interface OrderSubtotalsColumns {
    OrderId: Column<OrderSubtotalsRow>;
    Subtotal: Column<OrderSubtotalsRow>;
}

export class OrderSubtotalsColumns extends ColumnsBase<OrderSubtotalsRow> {
    static readonly columnsKey = 'Northwind.OrderSubtotals';
    static readonly Fields = fieldsProxy<OrderSubtotalsColumns>();
}

File: MovieProject/MovieProject.Web/Modules/Northwind/OrderSubtotals/OrderSubtotalsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.OrderSubtotals")]
[BasedOnRow(typeof(OrderSubtotalsRow), CheckNames = true)]
public class OrderSubtotalsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int OrderId { get; set; }
    public decimal Subtotal { get; set; }
}